libs/x-content/src/main/java/org/elasticsearch/common/xcontent/XContentSubParser.java

```diff
@@ -0,0 +1,280 @@
+    public Object objectText() throws IOException {
+        return parser.objectText();
+    }
+
+    @Override
+    public Object objectBytes() throws IOException {
+        return parser.objectBytes();
+    }
+
+    @Override
+    public boolean hasTextCharacters() {
+        return parser.hasTextCharacters();
+    }
+
+    @Override
+    public char[] textCharacters() throws IOException {
+        return parser.textCharacters();
+    }
+
+    @Override
+    public int textLength() throws IOException {
+        return parser.textLength();
+    }
+
+    @Override
+    public int textOffset() throws IOException {
+        return parser.textOffset();
+    }
+
+    @Override
+    public Number numberValue() throws IOException {
+        return parser.numberValue();
+    }
+
+    @Override
+    public NumberType numberType() throws IOException {
+        return parser.numberType();
+    }
+
+    @Override
+    public short shortValue(boolean coerce) throws IOException {
+        return parser.shortValue(coerce);
+    }
+
+    @Override
+    public int intValue(boolean coerce) throws IOException {
+        return parser.intValue(coerce);
+    }
+
+    @Override
+    public long longValue(boolean coerce) throws IOException {
+        return parser.longValue(coerce);
+    }
+
+    @Override
+    public float floatValue(boolean coerce) throws IOException {
+        return parser.floatValue(coerce);
+    }
+
+    @Override
+    public double doubleValue(boolean coerce) throws IOException {
+        return parser.doubleValue();
+    }
+
+    @Override
+    public short shortValue() throws IOException {
+        return parser.shortValue();
+    }
+
+    @Override
+    public int intValue() throws IOException {
+        return parser.intValue();
+    }
+
+    @Override
+    public long longValue() throws IOException {
+        return parser.longValue();
+    }
+
+    @Override
+    public float floatValue() throws IOException {
+        return parser.floatValue();
+    }
+
+    @Override
+    public double doubleValue() throws IOException {
+        return parser.doubleValue();
+    }
+
+    @Override
+    public boolean isBooleanValue() throws IOException {
+        return parser.isBooleanValue();
+    }
+
+    @Override
+    public boolean booleanValue() throws IOException {
+        return parser.booleanValue();
+    }
+
+    @Override
+    public byte[] binaryValue() throws IOException {
+        return parser.binaryValue();
+    }
+
+    @Override
+    public XContentLocation getTokenLocation() {
+        return parser.getTokenLocation();
+    }
+
+    @Override
+    public <T> T namedObject(Class<T> categoryClass, String name, Object context) throws IOException {
+        return parser.namedObject(categoryClass, name, context);
+    }
+
+    @Override
+    public NamedXContentRegistry getXContentRegistry() {
+        return parser.getXContentRegistry();
+    }
+
+    @Override
+    public boolean isClosed() {
+        return parser.isClosed();
```

Comment:
I'm wondering whether the fact that this would still return `false` after `close` has been called could be confusing.

Reply:
Indeed! I missed this one. Good catch! I will fix that.